
```{r gwas and signif_snps}
install.packages("data.table")
#install.packages("R.utils")
library(data.table)
#library(R.utils)
file_path <- "/Users/wqy/Downloads/GCST90454344.tsv"
gwas_data <- fread(file_path, sep = "\t", header = TRUE)
# 查看数据
str(gwas_data)
head(gwas_data)
colnames(gwas_data)
#尝试1kg
library(dplyr) 
install.packages("readr")
library(readr)   
bim_dir <- "/Users/wqy/Downloads/1KG"
colnames(gwas_data)[colnames(gwas_data) == "chromosome"] <- "CHROM"
colnames(gwas_data)[colnames(gwas_data) == "base_pair_location"] <- "POS"
#尝试1kg
bim_data <- data.frame()
for (chr in 1:22) {
  bim_file <- file.path(bim_dir, paste0("1000G_eur_chr", chr, ".bim"))
  if (file.exists(bim_file)) {
    # 读取 .bim 文件
    bim_chr <- read_table2(bim_file, col_names = c("CHROM", "rsID", "cm", "POS", "REF", "ALT"))
    bim_data <- bind_rows(bim_data, bim_chr)
  }
}
head(bim_data)
# match
annotated_gwas <- gwas_data %>%
  inner_join(bim_data, by = c("CHROM", "POS"))
head(annotated_gwas)
write_tsv(annotated_gwas, "/Users/wqy/Downloads/annotated_gwas_data.tsv")
#尝试1kg

significance_threshold <- 7.3  # 对应 p-value < 5e-8
significant_snps <- annotated_gwas %>%
  filter(neg_log_10_p_value >= significance_threshold)
print(dim(significant_snps))  # 显示显著 SNP 的数量
print(head(significant_snps)) 
# rsID
significant_snp_ids <- significant_snps$SNP
write.table(significant_snp_ids, "significant_snps.txt", row.names = FALSE, col.names = FALSE, quote = FALSE)
#图
library(ggplot2)
# 显著 SNP 分布图
significant_snps <- annotated_gwas[annotated_gwas$`neg_log_10_p_value` > 7.301, ]
ggplot(significant_snps, aes(x = CHROM, y = -log10(10^(-neg_log_10_p_value)))) +
  geom_point(alpha = 0.6) +
  labs(title = "Significant SNP Distribution",
       x = "Chromosome",
       y = "-log10(P-value)") +
  theme_minimal()
#图



```
```{r twas}


```

